### ISSUE TITLE
The old version of maven-war-plugin:2.2 is resolved

### ISSUE THREAD

**Issue by micobarac** [jhipster/generator-jhipster#10104]
##### **Overview of the issue**

I have noticed that an old version of **maven-war-plugin:2.2** is resolved, instead of the new one **maven-war-plugin:3.2.3**, as stated inside **pom.xml**.

```xml
<maven-war-plugin.version>3.2.3</maven-war-plugin.version>
```

```xml
<profile>
    <id>war</id>
    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-war-plugin</artifactId>
                <version>${maven-war-plugin.version}</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>war</goal>
                        </goals>
                        <phase>package</phase>
                    </execution>
                </executions>
                <configuration>
                    <warSourceIncludes>WEB-INF/**,META-INF/**</warSourceIncludes>
                    <failOnMissingWebXml>false</failOnMissingWebXml>
                </configuration>
            </plugin>
        </plugins>
    </build>
</profile>
```

![image](https://user-images.githubusercontent.com/16001837/61357046-d33bd880-a877-11e9-90d5-94f73ba6a8bd.png)

When I run:

`mvn help:effective-pom`

I get two dependencies for **maven-war-plugin**:

```xml
<profile>
  <id>war</id>
  <build>
    <plugins>
      <plugin>
        <artifactId>maven-war-plugin</artifactId>
        <version>3.2.3</version>
        <executions>
          <execution>
            <phase>package</phase>
            <goals>
              <goal>war</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <warSourceIncludes>WEB-INF/**,META-INF/**</warSourceIncludes>
          <failOnMissingWebXml>false</failOnMissingWebXml>
        </configuration>
      </plugin>
    </plugins>
  </build>
</profile>
```
 and

```xml
<plugin>
  <artifactId>maven-war-plugin</artifactId>
  <version>2.2</version>
  <executions>
    <execution>
      <id>default-war</id>
      <phase>package</phase>
      <goals>
        <goal>war</goal>
      </goals>
    </execution>
  </executions>
</plugin>
```

Where is this second one coming from?

##### **Motivation for or Use Case**

Resolve **maven-war-plugin:3.2.3**, as stated inside **pom.xml**.

##### **Reproduce the error**

Generate a new JHipster Cassandra project:

`jhipster --skip-client --yarn`

##### **Suggest a Fix**

Resolve **maven-war-plugin:3.2.3**, as stated inside **pom.xml**.

##### **JHipster Version(s)**

`6.1.2`

##### **JHipster Version(s)**

```
openprovider@0.0.0 /Users/milan/Sites/jhipster
└── generator-jhipster@6.1.2

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.logate.openprovider"
    },
    "jhipsterVersion": "6.1.2",
    "applicationType": "monolith",
    "baseName": "openprovider",
    "packageName": "com.logate.openprovider",
    "packageFolder": "com/logate/openprovider",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "cassandra",
    "devDatabaseType": "cassandra",
    "prodDatabaseType": "cassandra",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": false,
    "clientPackageManager": "yarn",
    "skipClient": true
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_202"
Java(TM) SE Runtime Environment (build 1.8.0_202-b08)
Java HotSpot(TM) 64-Bit Server VM (build 25.202-b08, mixed mode)

git version 2.21.0

node: v10.15.3

npm: 6.9.0

yeoman: 3.0.0

yarn: 1.16.0

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

##### **Browsers and Operating System**

macOS Mojave
Version 10.14.5 (18F132)

**Comment by gmarziou** [comment]
I used your .yo-rc.json and I could not reproduce with mvnw, so I suppose it comes from your environment, maybe the IDE.


<details>
<summary>mvnw dependency:resolve-plugins -Pwar</summary>
<pre>
[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< com.logate.openprovider:openprovider >----------------
[INFO] Building Openprovider 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:resolve-plugins (default-cli) @ openprovider ---
[INFO] Plugin Resolved: maven-deploy-plugin-2.7.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.0.6.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-1.5.6.jar
[INFO] Plugin Resolved: maven-failsafe-plugin-3.0.0-M3.jar
[INFO]     Plugin Dependency Resolved: maven-surefire-common-3.0.0-M3.jar
[INFO] Plugin Resolved: maven-clean-plugin-3.1.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-shared-utils-3.2.1.jar
[INFO] Plugin Resolved: maven-jar-plugin-2.4.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-archiver-2.5.jar
[INFO]     Plugin Dependency Resolved: plexus-archiver-2.1.jar
[INFO]     Plugin Dependency Resolved: commons-lang-2.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.jar
[INFO] Plugin Resolved: jib-maven-plugin-1.3.0.jar
[INFO]     Plugin Dependency Resolved: google-http-client-1.27.0.jar
[INFO]     Plugin Dependency Resolved: commons-compress-1.18.jar
[INFO]     Plugin Dependency Resolved: guava-27.0.1-jre.jar
[INFO]     Plugin Dependency Resolved: jackson-databind-2.9.9.jar
[INFO]     Plugin Dependency Resolved: asm-7.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.5.2.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.5.2.jar
[INFO]     Plugin Dependency Resolved: maven-verifier-1.6.jar
[INFO] Plugin Resolved: maven-surefire-plugin-3.0.0-M3.jar
[INFO]     Plugin Dependency Resolved: maven-surefire-common-3.0.0-M3.jar
[INFO] Plugin Resolved: spring-boot-maven-plugin-2.1.6.RELEASE.jar
[INFO]     Plugin Dependency Resolved: spring-boot-loader-tools-2.1.6.RELEASE.jar
[INFO]     Plugin Dependency Resolved: maven-archiver-3.2.0.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-3.3.9.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.3.9.jar
[INFO]     Plugin Dependency Resolved: maven-model-3.3.9.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.3.9.jar
[INFO]     Plugin Dependency Resolved: maven-settings-3.3.9.jar
[INFO]     Plugin Dependency Resolved: maven-common-artifact-filters-3.0.1.jar
[INFO]     Plugin Dependency Resolved: plexus-archiver-3.6.0.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.1.0.jar
[INFO]     Plugin Dependency Resolved: plexus-build-api-0.0.7.jar
[INFO]     Plugin Dependency Resolved: maven-shade-plugin-3.2.1.jar
[INFO] Plugin Resolved: maven-war-plugin-3.2.3.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-archiver-3.4.0.jar
[INFO]     Plugin Dependency Resolved: commons-io-2.5.jar
[INFO]     Plugin Dependency Resolved: plexus-archiver-4.1.0.jar
[INFO]     Plugin Dependency Resolved: plexus-interpolation-1.25.jar
[INFO]     Plugin Dependency Resolved: xstream-1.4.10.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.2.0.jar
[INFO]     Plugin Dependency Resolved: maven-filtering-3.1.1.jar
[INFO]     Plugin Dependency Resolved: maven-mapping-3.0.0.jar
[INFO] Plugin Resolved: maven-resources-plugin-3.1.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-model-3.0.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.1.0.jar
[INFO]     Plugin Dependency Resolved: maven-filtering-3.1.1.jar
[INFO]     Plugin Dependency Resolved: commons-io-2.5.jar
[INFO]     Plugin Dependency Resolved: plexus-interpolation-1.24.jar
[INFO] Plugin Resolved: maven-idea-plugin-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.0.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.0.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.0.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-manager-2.0.1.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.0.1.jar
[INFO]     Plugin Dependency Resolved: plexus-container-default-1.0-alpha-9.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-1.1.jar
[INFO]     Plugin Dependency Resolved: wagon-provider-api-1.0-alpha-5.jar
[INFO]     Plugin Dependency Resolved: dom4j-1.6.1.jar
[INFO] Plugin Resolved: jacoco-maven-plugin-0.8.4.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.22.jar
[INFO]     Plugin Dependency Resolved: file-management-1.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-impl-2.1.jar
[INFO]     Plugin Dependency Resolved: org.jacoco.agent-0.8.4-runtime.jar
[INFO]     Plugin Dependency Resolved: org.jacoco.core-0.8.4.jar
[INFO]     Plugin Dependency Resolved: org.jacoco.report-0.8.4.jar
[INFO] Plugin Resolved: maven-install-plugin-2.4.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-manager-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.0.6.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.5.jar
[INFO]     Plugin Dependency Resolved: plexus-digest-1.0.jar
[INFO] Plugin Resolved: properties-maven-plugin-1.0.0.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.22.jar
[INFO] Plugin Resolved: sonar-maven-plugin-3.6.0.1398.jar
[INFO]     Plugin Dependency Resolved: plexus-sec-dispatcher-1.4.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.22.jar
[INFO]     Plugin Dependency Resolved: sonar-scanner-api-2.12.0.1661.jar
[INFO]     Plugin Dependency Resolved: commons-lang-2.6.jar
[INFO] Plugin Resolved: maven-site-plugin-3.3.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-exec-1.1.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-model-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-settings-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-settings-builder-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-archiver-2.4.2.jar
[INFO]     Plugin Dependency Resolved: doxia-sink-api-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-logging-api-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-core-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-module-xhtml-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-module-apt-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-module-xdoc-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-module-fml-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-module-markdown-1.4.jar
[INFO]     Plugin Dependency Resolved: servlet-api-2.5.jar
[INFO]     Plugin Dependency Resolved: doxia-decoration-model-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-site-renderer-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-integration-tools-1.5.jar
[INFO]     Plugin Dependency Resolved: wagon-provider-api-1.0.jar
[INFO]     Plugin Dependency Resolved: plexus-archiver-1.0.jar
[INFO]     Plugin Dependency Resolved: plexus-i18n-1.0-beta-7.jar
[INFO]     Plugin Dependency Resolved: velocity-1.5.jar
[INFO]     Plugin Dependency Resolved: plexus-velocity-1.1.8.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-1.5.10.jar
[INFO]     Plugin Dependency Resolved: jetty-6.1.25.jar
[INFO]     Plugin Dependency Resolved: jetty-util-6.1.25.jar
[INFO]     Plugin Dependency Resolved: commons-lang-2.5.jar
[INFO]     Plugin Dependency Resolved: commons-io-1.4.jar
[INFO] Plugin Resolved: maven-javadoc-plugin-3.1.0.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-model-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-settings-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-archiver-3.2.0.jar
[INFO]     Plugin Dependency Resolved: maven-invoker-3.0.0.jar
[INFO]     Plugin Dependency Resolved: maven-common-artifact-filters-3.0.0.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-transfer-0.10.1.jar
[INFO]     Plugin Dependency Resolved: doxia-sink-api-1.7.jar
[INFO]     Plugin Dependency Resolved: doxia-site-renderer-1.7.4.jar
[INFO]     Plugin Dependency Resolved: wagon-provider-api-1.0-beta-6.jar
[INFO]     Plugin Dependency Resolved: commons-lang3-3.5.jar
[INFO]     Plugin Dependency Resolved: commons-io-2.5.jar
[INFO]     Plugin Dependency Resolved: httpclient-4.5.2.jar
[INFO]     Plugin Dependency Resolved: qdox-2.0-M10.jar
[INFO]     Plugin Dependency Resolved: plexus-java-1.0.3.jar
[INFO]     Plugin Dependency Resolved: plexus-archiver-3.6.0.jar
[INFO]     Plugin Dependency Resolved: plexus-io-3.1.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.24.jar
[INFO]     Plugin Dependency Resolved: plexus-interactivity-api-1.0-alpha-6.jar
[INFO] Plugin Resolved: maven-enforcer-plugin-3.0.0-M1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.24.jar
[INFO]     Plugin Dependency Resolved: enforcer-api-3.0.0-M1.jar
[INFO]     Plugin Dependency Resolved: enforcer-rules-3.0.0-M1.jar
[INFO] Plugin Resolved: maven-eclipse-plugin-2.10.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-manager-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-core-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-settings-2.2.1.jar
[INFO]     Plugin Dependency Resolved: wagon-provider-api-2.1.jar
[INFO]     Plugin Dependency Resolved: commons-io-2.2.jar
[INFO]     Plugin Dependency Resolved: plexus-archiver-2.6.3.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.20.jar
[INFO]     Plugin Dependency Resolved: plexus-interactivity-jline-1.0-alpha-5.jar
[INFO]     Plugin Dependency Resolved: maven-archiver-2.5.jar
[INFO]     Plugin Dependency Resolved: plexus-resources-1.0-alpha-7.jar
[INFO]     Plugin Dependency Resolved: bndlib-0.0.145.jar
[INFO]     Plugin Dependency Resolved: maven-osgi-0.2.0.jar
[INFO]     Plugin Dependency Resolved: resources-3.3.0-v20070604.jar
[INFO] Plugin Resolved: maven-compiler-plugin-3.8.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-shared-utils-3.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-shared-incremental-1.1.jar
[INFO]     Plugin Dependency Resolved: plexus-java-0.9.10.jar
[INFO]     Plugin Dependency Resolved: plexus-compiler-api-2.8.4.jar
[INFO]     Plugin Dependency Resolved: plexus-compiler-manager-2.8.4.jar
[INFO]     Plugin Dependency Resolved: plexus-compiler-javac-2.8.4.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.271 s
[INFO] Finished at: 2019-07-17T22:11:27+02:00
[INFO] ------------------------------------------------------------------------
</pre>
</details>


And here is the effective pom: `mvnw -Pwar help:effective-pom`

[eff_pom.xml.txt](https://github.com/jhipster/generator-jhipster/files/3403718/eff_pom.xml.txt)

**Comment by micobarac** [comment]
Hmmm, after running the `mvnw dependency:resolve-plugins -Pwar` command, the `mvnw -Pwar help:effective-pom` shows only the version noted inside pom.xml. But, the Intellij Idea still shows the old version of plugin inside `Maven -> Project name -> Plugins`:

![image](https://user-images.githubusercontent.com/16001837/61357046-d33bd880-a877-11e9-90d5-94f73ba6a8bd.png)

So, this issue obviously has nothing to do with JHIpster.

Anyway, thanks for your support. :)
